geode-for-redis/src/distributedTest/java/org/apache/geode/redis/internal/commands/executor/list/LPopDUnitTest.java

```diff
@@ -144,18 +144,19 @@ private void lpopPerformAndVerify(String key, AtomicLong runningCount) {
 
     int elementCount = INITIAL_LIST_SIZE - 1;
     while (jedis.llen(key) > 0 && runningCount.get() > 0) {
-      String expected = "-" + key + "-" + (elementCount) + "-";
+      String expected = makeElementString(key, elementCount);
       String element = expected;
       try {
         element = jedis.lpop(key);
-        if (element.equals(expected)) {
-          elementCount--;
-        } else {
-          int expectedCount = elementCount - 1;
-          String[] chunks = element.split("-");
-          elementCount = Integer.parseInt(chunks[4]);
-          assertThat(expectedCount - elementCount).isLessThanOrEqualTo(1);
-        }
+        String[] chunks = element.split("-");
+        int actualCount = Integer.parseInt(chunks[4]);
+        int expectedCount = elementCount;
+        assertThat(actualCount)
+            .as("Actual element count did not match expected element count for element " + element)
+            .satisfiesAnyOf(
+                count -> assertThat(count).isEqualTo(expectedCount),
+                count -> assertThat(count).isEqualTo(expectedCount + 1));
```

Comment:
This should be `.isEqualTo(expectedCount - 1)` since when doing LPOP, we're decrementing the element count each time, so a duplicated LPOP should result in an element count that's one less than expected.